DocGen4/Output.lean

```diff
@@ -20,6 +22,16 @@ namespace DocGen4
 
 open Lean IO System Output Process
 
+def collectBackrefs : IO (Array BackrefItem) := do
+  let mut backrefs : Array BackrefItem := #[]
+  for entry in ← System.FilePath.readDir declarationsBasePath do
+    if entry.fileName.startsWith "backrefs-" && entry.fileName.endsWith ".json" then
+      let fileContent ← FS.readFile entry.path
+      let .ok jsonContent := Json.parse fileContent | unreachable!
```

Comment:
If we ever violate one of these invariants Lean is only going to give us an "unreachable code reached" error message. Given the fact that you are in `IO` here you can throw a proper error so we can actually debug this kind of stuff in the future. Dito for other calls of `unreachable!` in monadic contexts.

Reply:
Maybe using `IO.ofExcept` here is reasonable?

Reply:
Great, you can also change it there if you want :+1:

Reply:
Done.